src/Text/Layout/Table/Primitives/ColumnModifier.hs

```diff
@@ -149,15 +156,32 @@ deriveColModInfoFromColumnLA (lenS, alignS) = case alignS of
                 ExpandBetween i j -> expandBetween' i j
         in interpretLenSpec . measureMaximumWidth
 
+    -- Both the original length and requested length of an object
+    requestedLength a = RequestedLength l $ case lenS of
+        Fixed i                   -> i
+        ExpandUntil i | l > i     -> i - paddingNeeded i a
+        FixedUntil i              -> max i l
+        ExpandBetween i j | l > j -> max i $ j - paddingNeeded i a
+        ExpandBetween i _ | l < i -> i
+        _                         -> l
+      where
+        l = visibleLength a
+
+        paddingNeeded i = totalAdjustment' . buildCellViewTight . case posS of
```

Comment:
What I don't like about this is that it uses more than just functions for measuring from `Cell`.  This looks like a code smell to me and makes me wonder whether there is not a more direct approach.  For example, could `Cell` have a function that does this:
```
visibleLengthAfterDrop
    :: Cell a
    => a
    -> Int -- ^ The amount to drop left.
    -> Int -- ^ The amount to drop right.
    -> Int -- ^ The actual visible width of the remaining characters.
visibleLengthAfterDrop c l r = visibleLength c - (l + r) -- default implementation
```
Instances that support multi-width characters can then return a lesser amount.

Reply:
I agree that this is a code smell, but I think the solution goes in the other direction. At this point we know exactly what we need to trim to, and past this point the only possibility is that we need to add more padding. We should be keeping the trimmed result around along with its length, and then padding based on the other elements in the column afterwards.

Reply:
> We should be keeping the trimmed result around along with its length, and then padding based on the other elements in the column afterwards.

This function has the purpose of getting an abstract description of what to do with columns.  One example why this is required is the headers.  Thus, I want to keep this part unless there is a good reason to change it, and I don't see that at the moment.

The only drawback of having something like `visibleLengthAfterDrop` that I can see is the duplicate wide character scanning.  Could you explain why this would **not** work?  What am I missing here?

Reply:
It's not that it won't work: it will. The reason I'm reluctant to switch is that calculating the visible length after drop and building the result of dropping are almost exactly the same work. Implementing the two things separately while avoiding code duplication would result in writing a function which returns both as a tuple, and then implementing the two typeclass functions as projecting that tuple to one or the other. This forces you to calculate things twice. By returning both at the same time, you at least have the option of reusing the work.